test/python/quantum_info/test_synthesis.py

```diff
@@ -1542,6 +1546,43 @@ def test_opt_a1a2(self, nqubits):
             ccirc.count_ops().get("cx"), (23 / 48) * 4**nqubits - (3 / 2) * 2**nqubits + 4 / 3
         )
 
+    def test_cs_decomp_error_raises(self):
+        """
+        Test that either an exception is raised or the decomposition succeeds.
+        """
+        base_gate_dim = 4
+        num_ctrls = 4
+        for i in range(3, 5):  # on my system i=3 induced exception but may vary
+            base_gate = random_unitary(base_gate_dim, seed=i * 2089).data
+            cmat = _compute_control_matrix(base_gate, num_ctrls)
+            try:
+                qc = self.qsd(cmat)
+            except QiskitError as err:
+                if "CS decomposition error" in repr(err):
+                    pass
+                else:
+                    self.fail("unexpected QiskitError")
+            else:
+                cqcop = Operator(qc)
+                self.assertTrue(cqcop == Operator(cmat))
```

Comment:
I'm not sure what this test is trying to do here. Once the error is a custom type, I think you might be meaning:
```python
try:
    qc = self.qsd(cmat)
    # Either the decomposition succeeds ...
    self.assertEqual(Operator(qc), Operator(cmat))
except QSDError:
    # ... or the correct type of error is raised.
    pass
```

Reply:
I was attempting to test for this specific scipy cossin decomposition error, as opposed to other errors which might be generated by the function. However, since the occurrence seems to be sensitive to software installation (the input matrix you mentioned above worked fine for me but others did not), I am not sure how to guarantee that this test will hit the error, in which case it may be ok to just remove it?

Reply:
Yeah, if we can't reliably reproduce it, it's _probably_ best just skip the test and just open it as an issue on Terra instead.